### PR TITLE
Meta: Update pre-commit to fix Sphinx Lint

### PR DESCRIPTION
**Pull request by hugovk**
Re: https://github.com/sphinx-contrib/sphinx-lint/pull/106

Also update codespell and other hooks.

### CODE REVIEW

File: .codespell/ignore-words.txt [hunk 1]
```diff
@@ -1,19 +1,27 @@
 adaptee
 ancilliary
 ans
+anull
```
Comment (JelleZijlstra) [inline .codespell/ignore-words.txt:4]: Just curious, why do we need this? `git grep` didn't find this word

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -42,8 +42,8 @@ repos:
       - id: check-yaml
         name: "Check YAML"
 
-  - repo: https://github.com/psf/black
-    rev: 23.7.0
+  - repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 23.11.0
```
Comment (JelleZijlstra) [inline .pre-commit-config.yaml:46]:
```suggestion
    rev: 23.12.0
```

File: .codespell/ignore-words.txt (hunk 1 above)
Comment (hugovk) [reply .codespell/ignore-words.txt:4]: Mixed caps :)

```
peps/pep-0543.rst:838: aNULL ==> annul
```
```console
❯ git grep anull

❯ git grep -i anull
peps/pep-0543.rst:    'ECDH+AESGCM:ECDH+CHACHA20:DH+AESGCM:DH+CHACHA20:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:RSA+AESGCM:RSA+AES:!aNULL:!eNULL:!MD5'
```

Comment (JelleZijlstra) [reply .codespell/ignore-words.txt:4]: Ah, didn't think of that!

**Review by JelleZijlstra — APPROVED** [review]

**Comment by 1atAlcone** [comment]
#

**Pull request merged**